### PR TITLE
Improving barline rendition

### LINKED ISSUE #1321: Dots missing on repeat barlines encoded with <barLine>
This:

```xml
    <barLine form="rptstart"/>
    <rest dur="4"/>
    <barLine form="rptboth"/>
    <rest dur="4"/>
    <barLine form="rptend"/>
    <rest dur="4"/>
```

Becomes this.

grafik

[repeat.mei.zip](https://github.com/rism-ch/verovio/files/4248480/repeat.mei.zip)

### LINKED ISSUE #1333: dots from repeats lose alingment on scaling
The repeat dots on scaled barlines are positioned wrong. 

score_001

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title></title>
            </titleStmt>
            <pubStmt>
                <date></date>
            </pubStmt>
        </fileDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef label="feature-example">
                      <meterSig unit="4" count="4"/>
                        <staffGrp>
                            <staffGrp symbol="bracketsq" bar.thru="true" xml:id="System">
                                <staffDef n="1" clef.line="2" clef.shape="G" lines="5" scale="75%"/>
                                <staffGrp symbol="brace" xml:id="klammern">
                                    <staffDef n="2" clef.line="2" clef.shape="G" lines="5"/>
                                    <staffDef n="3" clef.line="2" clef.shape="G" lines="5"/>
                                </staffGrp>
                            </staffGrp>
                            <staffDef n="4" clef.line="2" clef.shape="G" lines="5" scale="50%"/>
                        </staffGrp>
                    </scoreDef>
                    <section>
                        <measure n="4" right="rptstart">
                            <staff n="1">
                                <layer n="1"><mRest/></layer>
                            </staff>
                            <staff n="2">
                                <layer n="1"><mRest/></layer>
                            </staff>
                            <staff n="3">
                                <layer n="1"><mRest/></layer>
                            </staff>
                            <staff n="4">
                                <layer n="1"><mRest/></layer>
                            </staff>
                        </measure>
                        <measure n="5" right="rptboth">
                            <staff n="1">
                                <layer n="1"><mRest/></layer>
                            </staff>
                            <staff n="2">
                                <layer n="1"><mRest/></layer>
                            </staff>
                            <staff n="3">
                                <layer n="1"><mRest/></layer>
                            </staff>
                            <staff n="4">
                                <layer n="1"><mRest/></layer>
                            </staff>
                        </measure>
                        <measure n="6" right="rptend">
                            <staff n="1">
                                <layer n="1"><mRest/></layer>
                            </staff>
                            <staff n="2">
                                <layer n="1"><mRest/></layer>
                            </staff>
                            <staff n="3">
                                <layer n="1"><mRest/></layer>
                            </staff>
                            <staff n="4">
                                <layer n="1"><mRest/></layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

Not connected to #1321

### PR DESCRIPTION
**Pull request by rettinghaus**
Adds full support for SMuFLs `barlineSeparation` and `repeatBarlineDotSeparation`
closes #1321
closes #1333

Issue #1333
barline

Issue #1321 
barline

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0" ?>
<?xml-model href="https://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron" ?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
  <meiHead>
    <fileDesc>
      <titleStmt>
        <title>Repeated barLine elements</title>
      </titleStmt>
      <pubStmt>
        <respStmt>
          <persName role="encoder">Klaus Rettinghaus</persName>
          <corpName role="funder">eNote GmbH</corpName>
        </respStmt>
        <date isodate="2020">2020</date>
      </pubStmt>
      <seriesStmt>
        <title>Verovio test suite</title>
      </seriesStmt>
      <notesStmt>
        <annot>Verovio renders all barLine elements even on single line staves.</annot>
      </notesStmt>
    </fileDesc>
    <encodingDesc>
      <appInfo>
        <application version="3.0.0" label="1">
          <name>Verovio</name>
        </application>
      </appInfo>
    </encodingDesc>
  </meiHead>
  <music>
    <body>
      <mdiv>
        <score>
          <scoreDef>
            <staffGrp>
              <staffDef n="1" lines="1" clef.shape="perc" meter.count="3" meter.unit="4" />
            </staffGrp>
          </scoreDef>
          <section>
            <measure>
              <staff n="1">
                <layer n="1">
                  <rest dur="4" />
                  <barLine form="rptstart" />
                  <tuplet num="3" numbase="2">
                    <note loc="0" stem.dir="up" dur="4" />
                    <note loc="0" stem.dir="up" dur="4" />
                    <note loc="0" stem.dir="up" dur="4" />
                  </tuplet>
                </layer>
              </staff>
            </measure>
            <measure>
              <staff n="1">
                <layer n="1">
                  <tuplet num="3" numbase="2">
                    <note loc="0" stem.dir="up" dur="4" />
                    <note loc="0" stem.dir="up" dur="4" />
                    <note loc="0" stem.dir="up" dur="4" />
                  </tuplet>
                  <barLine form="rptboth" />
                  <rest dur="4" />
                </layer>
              </staff>
            </measure>
            <measure>
              <staff n="1">
                <layer n="1">
                  <tuplet num="3" numbase="2">
                    <note loc="0" stem.dir="up" dur="4" />
                    <note loc="0" stem.dir="up" dur="4" />
                    <note loc="0" stem.dir="up" dur="4" />
                  </tuplet>
                  <barLine form="rptend" />
                  <rest dur="4" />
                </layer>
              </staff>
            </measure>
          </section>
        </score>
      </mdiv>
    </body>
  </music>
</mei>
```

I'll add test files to the test suite.

### CODE REVIEW

File: src/options.cpp [hunk 1]
```diff
@@ -769,7 +769,11 @@ Options::Options()
     m_generalLayout.SetLabel("General layout options", "2-generalLayout");
     m_grps.push_back(&m_generalLayout);
 
-    m_barLineWidth.SetInfo("Bar line width", "The barLine width");
+    m_barlineSeparation.SetInfo("Barline separation", "The default distance between multiple barlines when locked together");
+    m_barlineSeparation.Init(0.8, 0.5, 2.0);
+    this->Register(&m_barlineSeparation, "barlineSeparation", &m_generalLayout);
```
Comment (lpugin) [inline src/options.cpp:774]: Verovio spelling follows MEI (`barLine`)

Comment (rettinghaus) [reply src/options.cpp]: hopefully solved.

**Comment by craigsapp** [comment]
Looks good, although the #1333 example target rendering would look better like this:

Screen Shot 2020-10-06 at 7 06 06 AM

The barline that attaches multiple stafflines should be the same size (of the largest staff) across all staves as your rendering does.  But for unattached barlines, they should match the size of the staff on which they are displayed (bottom staff).

Probably unrelated, but why are the rests in the top staff centered half-rests?  The example data has `<mRest>` so the rendering of the rests on the top staff are incorrect.

**Comment by rettinghaus** [comment]
See #1728

**Pull request merged**